src/Kernel/Magic/Simulate.cs

```diff
@@ -94,12 +97,26 @@ public async Task<ExecutionResult> RunAsync(string input, IChannel channel)
             var symbol = SymbolResolver.Resolve(name) as IQSharpSymbol;
             if (symbol == null) throw new InvalidOperationException($"Invalid operation name: {name}");
 
+            var maxNQubits = 0L;
+
             using var qsim = new QuantumSimulator()
                 .WithJupyterDisplay(channel, ConfigurationSource)
                 .WithStackTraceDisplay(channel);
             qsim.OnDisplayableDiagnostic += channel.Display;
+            qsim.OnAllocateQubits += nQubits =>
+            {
+                maxNQubits = System.Math.Max(qsim.QubitManager.GetAllocatedQubitsCount(), maxNQubits);
+            };
+            var stopwatch = new Stopwatch();
             var value = await symbol.Operation.RunAsync(qsim, inputParameters);
-            return value.ToExecutionResult();
+            stopwatch.Stop();
+            var result = value.ToExecutionResult();
+            (Monitor as PerformanceMonitor)?.ReportSimulatorPerformance(new SimulatorPerformanceArgs(
+                simulatorName: typeof(SimulateMagic).FullName,
```

Comment:
Would it make sense to report the name of the simulator itself (e.g., `qsim.GetType().FullName`) rather than the name of the magic command?

Reply:
Good catch, thanks; I'll modify accordingly. That should also be a lot easier to use if we generalize how we handle different simulators in the future.